homeassistant/components/websocket_api/messages.py

```diff
@@ -65,12 +65,27 @@ def construct_result_message(iden: int, payload: str) -> str:
     return f'{{"id":{iden},"type":"result","success":true,"result":{payload}}}'
 
 
-def error_message(iden: int | None, code: str, message: str) -> dict[str, Any]:
+def error_message(
+    iden: int | None,
+    code: str,
+    message: str,
+    translation_key: str | None = None,
+    translation_domain: str | None = None,
+    translation_placeholders: dict[str, Any] | None = None,
+) -> dict[str, Any]:
     """Return an error result message."""
+    error_payload: dict[str, Any] = {
+        "code": code,
+        "message": message,
+    }
+    if translation_key is not None:
```

Comment:
Why do we need to guard?

Reply:
May be we don't need to do that

Reply:
Tried to remove it, but reverted that, it seems a lot of CI tests assert on the whole payload. If we leave the guard, we need to adjust all the tests.
I suggest that if this is needed we do this in a different PR.

Reply:
We shouldn't complicate the code to avoid updating tests. Check with frontend developers if they prefer to have the keys always there or only there when needed.
There's no need consider saving data here IMHO since sending error responses is not the common case.

Reply:
Right, I do not think it matters in the frontend, as there is no explit typing for the error responses. But we need to update quite a few tests. I would prefer to remove the guard in a different PR to avoid this one to become even bigger.
I asked Paul, if he says we should not add it, then we do not need to change is, but I assume he'll say that it does not matter.

Reply:
I had a discussion with @piitaya on Discord. He believes it is the right way to by only adding the translation attributes when there translation information is defined/available for the exception.